lib/resolve.ts

```diff
@@ -104,3 +111,23 @@ export function resolveTask(
 
 	return task;
 }
+
+/**
+ * Given a balena architecture string, translate it to the equivalent
+ * docker platform string.
+ */
+function resolveDockerPlatform(balenaArchitecture: string): string | null {
```

Comment:
It would be good to export this function, and further re-export it in `index.ts`:
```
export { resolveDockerPlatform } from './resolve';
```
The reason being that I found myself duplicating the logic in the CLI in order to conditionally print warning messages regarding the `--pull` and `--emulated` options when the app architecture doesn't match the docker daemon / balenaEngine host architecture.